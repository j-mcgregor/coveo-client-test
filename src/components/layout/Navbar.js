import React from 'react';
import { Link } from 'react-router-dom';
import S from '../../static/styles';
import Logo from '../../static/images/arrow.png';

const Navbar = () => (
  <S.Navbar className="navbar container grid-lg">
    <section className="navbar-section">
      <Link to="/" className="navbar-brand mr-2">
        {' '}
        <img src={Logo} alt="logo" width="20" />{' '}
      </Link>
      <Link to="/catalog/beers" className="btn btn-link">
        Beers
      </Link>
      <Link to="/catalog/wines" className="btn btn-link">
        Wines
      </Link>
      <Link to="/catalog/spirits" className="btn btn-link">
        Spirits
      </Link>
    </section>
    <section className="navbar-section">
      <div className="input-group input-inline">
        <input className="form-input" type="text" placeholder="search" />
        <button className="btn btn-primary input-group-btn" type="button">
          Search
        </button>
      </div>
    </section>
  </S.Navbar>
);

export default Navbar;
