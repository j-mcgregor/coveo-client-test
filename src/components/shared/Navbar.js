import React from 'react';
import { Link } from 'react-router-dom';
import S from '../../static/styles';
import createKey from '../../utils/createKey';

const options = [
  { to: 'beers', query: 'Beer' },
  { to: 'red-wine', query: 'Red wine' },
  { to: 'white-wine', query: 'White wine' }
];

const Navbar = () => (
  <S.Navbar className="navbar container grid-lg">
    <section className="navbar-section">
      <div className="dropdown">
        <div className="btn-group">
          <a href="#" className="btn dropdown-toggle" tabIndex="0">
            <span className="mr-2">More</span>
            <i className="fas fa-caret-down" />
          </a>
          <ul className="menu">
            {options.map((option, i) => (
              <li className="menu-item" key={createKey(option.query, i)}>
                <Link to={`/catalog/${option.to}`}>{option.query}</Link>
              </li>
            ))}
          </ul>
        </div>
      </div>
    </section>
  </S.Navbar>
);

export default Navbar;
