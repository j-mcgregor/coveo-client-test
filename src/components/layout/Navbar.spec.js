import React from 'react';
import { shallow } from 'enzyme';
import Navbar from './Navbar';

describe('Navbar', () => {
  it('should render the Navbar', () => {
    const wrapper = shallow(<Navbar />);

    expect(wrapper).toMatchSnapshot();
  });
});
