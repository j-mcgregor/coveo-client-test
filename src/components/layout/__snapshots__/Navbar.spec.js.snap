// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should not show the search bar if the root is "/" 1`] = `ReactWrapper {}`;

exports[`Navbar should render the Navbar 1`] = `ReactWrapper {}`;
